crates/interledger-store-redis/src/crypto.rs

```diff
@@ -6,28 +6,107 @@ use ring::{
 };
 
 const NONCE_LENGTH: usize = 12;
-static HMAC_KEY_GENERATION_STRING: &[u8] = b"ilp_store_redis_hmac_key";
 static ENCRYPTION_KEY_GENERATION_STRING: &[u8] = b"ilp_store_redis_encryption_key";
 
-pub fn generate_keys(
-    server_secret: &[u8],
-) -> (hmac::SigningKey, aead::SealingKey, aead::OpeningKey) {
-    let generation_key = hmac::SigningKey::new(&digest::SHA256, server_secret);
-    let hmac_key = hmac::SigningKey::new(
-        &digest::SHA256,
-        hmac::sign(&generation_key, HMAC_KEY_GENERATION_STRING).as_ref(),
-    );
-    let encryption_key = aead::SealingKey::new(
-        &aead::AES_256_GCM,
-        hmac::sign(&generation_key, ENCRYPTION_KEY_GENERATION_STRING).as_ref(),
-    )
-    .unwrap();
-    let decryption_key = aead::OpeningKey::new(
-        &aead::AES_256_GCM,
-        hmac::sign(&generation_key, ENCRYPTION_KEY_GENERATION_STRING).as_ref(),
-    )
-    .unwrap();
-    (hmac_key, encryption_key, decryption_key)
+use core::sync::atomic;
+use secrecy::{DebugSecret, Secret};
+use std::ptr;
+use zeroize::Zeroize;
+
+#[derive(Debug)]
+pub struct EncryptionKey(pub(crate) aead::SealingKey);
+
+#[derive(Debug)]
+pub struct DecryptionKey(pub(crate) aead::OpeningKey);
+
+#[derive(Debug)]
+pub struct GenerationKey(pub(crate) hmac::SigningKey);
+
+impl DebugSecret for EncryptionKey {}
+impl DebugSecret for DecryptionKey {}
+impl DebugSecret for GenerationKey {}
+
+impl Zeroize for EncryptionKey {
+    fn zeroize(&mut self) {
+        // Instead of clearing the memory, we create a key from an empty
```

Comment:
It's not empty -- empty suggests a zero-length (the other copies of this comment should be updated too)

```suggestion
        // Instead of clearing the memory, we create overwrite the key with
```